core/src/test/java/org/kiwix/kiwixmobile/core/utils/ExternalLinkOpenerTest.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.utils
+
+import android.app.Activity
+import android.content.Intent
+import android.widget.Toast
+import io.mockk.every
+import io.mockk.justRun
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.toast
+
+internal class ExternalLinkOpenerTest {
+  private val sharedPreferenceUtil: SharedPreferenceUtil = mockk()
+  private val alertDialogShower: AlertDialogShower = mockk(relaxed = true)
+  private val intent: Intent = mockk()
+  private val activity: Activity = mockk()
+
+  @Test
+  internal fun `alertDialogShower opens link if confirm-button is clicked`() {
+    every { intent.resolveActivity(activity.packageManager) } returns mockk()
+    every { sharedPreferenceUtil.prefExternalLinkPopup } returns true
+    val lambdaSlot = slot<() -> Unit>()
+    val externalLinkOpener =
+      spyk(ExternalLinkOpener(activity, sharedPreferenceUtil, alertDialogShower))
+    externalLinkOpener.openExternalUrl(intent)
+    verify {
+      alertDialogShower.show(any(), capture(lambdaSlot), any(), any())
```

Comment:
I only became aware of this recently and haven't had a chance to play around with it but this [in progress guidebook](https://notwoods.github.io/mockk-guidebook/docs/matching/with/) shows how to use `withArg` which can get rid of cumbersome capture/slot constructs, I don't know if you can verify inside a verify? might be fun to play with though and see what happens